### PR TITLE
Crimre 153 resubmission history

### PR DESCRIPTION
**Pull request by arthurashman**
## Description of change
Recursively iterate through parent ids to collate all application history items
Built history for all items
Link to previous superseded versions via respective history application received items

## Link to relevant ticket
https://dsdmoj.atlassian.net/browse/CRIMRE-153

## Screenshots of changes (if applicable)

image
image

## How to manually test the feature
Submit from apply
Assign to self, and return in review
Resubmit from apply
View application and application history

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@
   coverage_criterion :branch
   # primary_coverage :branch
   minimum_coverage 100
-  add_group 'Views', 'app/views'
+  add_filter 'app/views'
```
Comment (arthurashman) [inline spec/spec_helper.rb:11]: To fix readout on coverage output on rspec

File: app/models/crime_application.rb [hunk 2]
```diff
@@ -43,6 +43,19 @@ def find(id)
     end
   end
 
+  def parent
+    return nil if parent_id.nil?
+
+    CrimeApplication.find(parent_id)
```
Comment (timpeat) [inline app/models/crime_application.rb:49]: We should memoize this. Will do and push now.

File: spec/spec_helper.rb (hunk 1 above)
Comment (willmcb) [reply spec/spec_helper.rb:11]: Is this flagging that the view isn't covered? If so should we not add tests rather than a filter?

**Review by timpeat — APPROVED** [review]

**Review by willmcb — APPROVED** [review]
Nice one - I have added a few comments to consider - please have a look at let me know if you would like to chat anything through 👍

File: app/models/crime_application.rb [hunk 3]
```diff
@@ -43,6 +43,19 @@ def find(id)
...
+  end
+
+  def all_histories
+    histories = [history]
```
[the remark this answers is not in the document]
Comment (timpeat) [reply app/models/crime_application.rb:55]: Parent id is set by the datastore. But you make a REALLY good point.

File: app/views/crime_applications/history.html.erb [hunk 4]
```diff
@@ -25,8 +25,10 @@
       </tr>
     </thead>
     <tbody class="govuk-table__body">
-      <% @crime_application.history.items.each do |item| %>
-        <%= render partial: item, layout: 'history_item', as: :item, locals: { application: @crime_application } %>
+      <% @crime_application.all_histories.each do |history| %>
+        <% history.items.each do |item| %>
+          <%= render partial: item, layout: 'history_item', as: :item, locals: { current_page_application_id: @crime_application.id, application: history.application } %>
```
[the remark this answers is not in the document]
Comment (timpeat) [reply app/views/crime_applications/history.html.erb:30]: @willmcb this won't work as we need the application specific to the history here: 
```
application: history.application
```

File: spec/spec_helper.rb (hunk 1 above)
Comment (timpeat) [reply spec/spec_helper.rb:11]: Testing ERB with simple coverage is very new and has some issues. We can re-enable once that has been fixed.

**Pull request merged**